utopia-remix/app/routes/loaders/editorLoader.server.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import type { LoaderFunctionArgs } from '@remix-run/node'
+import { ALLOW, validateProjectAccess } from '../../handlers/validators'
+import { getUser, handle } from '../../util/api.server'
+import { proxy } from '../../util/proxy.server'
+import { UserProjectPermission } from '~/types'
+import { useLoaderData, type Params, json } from '@remix-run/react'
+import type { ApiError } from '~/util/errors'
+import { Status } from '~/util/statusCodes'
+import ProjectNotFound from '../../components/projectNotFound'
+
+export async function loader(args: LoaderFunctionArgs) {
+  return handle(args, {
+    GET: {
+      handler: getProjectForEditor,
+      validator: ALLOW,
+    },
+  })
+}
+
+const validator = validateProjectAccess(UserProjectPermission.CAN_VIEW_PROJECT, {
+  canRequestAccess: true,
+  getProjectId: (params) => params.id?.split('-')[0],
```

Comment:
Probably worth extracting out this logic into a function, I can see the same logic being used 3 lines later.